### ISSUE TITLE
"anchor-has-content" false negative if children is passed using spread syntax

### ISSUE THREAD

**Issue by lauriii** [evcohen/eslint-plugin-jsx-a11y#310]
I'm not sure if this is won't fix since this is kind of a weird edge case. We have following code in our project which I think should fulfill the a11y requirements since it will always have children, but the linter doesn't recognize it and gives a false negative.

```
// Use <a> element instead of button if button is a link.
if (typeof props.to !== 'undefined' && typeof props.children !== 'undefined') {
  return <Link {...props} />;
}
```

**Comment by ljharb** [comment]
Can you do `<Link {...props} to={props.to} children={props.children} />`? That way, the props are explicit, and static analysis tools like this linter plugin can detect them.

**Comment by lauriii** [comment]
This is indeed how we used to workaround this bug but we run into some proptype validation issues. For that reason, I wanted to remove the explicit passing of the props.children to not have to deal with the proptype validation.

**Comment by ljharb** [comment]
At Airbnb we've discovered that spread props are largely an antipattern, so we explicitly define propTypes and props everywhere we can. I'm not sure how you can get around either a) explicitly listing props, or b) overriding the rule in this case.
